### PR TITLE
Implement Approval Orb

### PR DESCRIPTION
**Pull request by llam15**
# Summary
- Implements the commands
   - `approval/request`
	   - Makes a POST request to `/events/request`
   - `approval/deescalate`
	   - Makes a POST request to `/events/deescalate`

- Implements the jobs
	- `approval/request`
		- Calls the `approval/request` command
		- Then persists the output to the workspace (if `request_slug` is provided)
	- `approval/deescalate`
		- Calls the `approval/deescalate` command
		- Can be used to de-escalate a request from the same workflow if given the `request_slug`

### CODE REVIEW

**Review by druiz — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
-A project template for Orbs.
+## Usage
+### Setup
+In order to use the Sym API Orb on CircleCI you will need to create a `SYM_JWT` and set it as an environment variable
+available for use in your pipeline.
 
-This repository is designed to be automatically ingested and modified by the CircleCI CLI's `orb init` command.
+1. Create a Bot user with `symflow bots create [bot-name]`
+2. Issue a `SYM_JWT` with `symflow tokens issue -u [bot-name] -e [expiry-length]`
+3. Save the JWT outputted by this previous step into an environment variable available to your workflow. You can do this by [adding the variable to a CircleCI Context](https://circleci.com/docs/2.0/env-vars#setting-an-environment-variable-in-a-context),
```
Comment (Arianna2028) [inline README.md:17]: I know we're referring to it as "a `SYM_JWT`" but we should probably be explicit about the fact that the env var _needs_ to be called `SYM_JWT`

File: src/commands/request.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+parameters:
+  flow_srn:
+    type: string
+    description: "The SRN of the Flow to trigger"
+  flow_inputs:
+    type: string
+    description: "The JSON-encoded flow_inputs to include in the request"
+  flow_context:
+    type: string
+    description: "The Json-encoded context to include in the request"
```
Comment (Arianna2028) [inline src/commands/request.yml:14]: nit: consistency
```suggestion
    description: "The JSON-encoded context to include in the request"
```

File: src/jobs/deescalate.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+description: >
+  Makes a Sym Request.
```
Comment (Arianna2028) [inline src/jobs/deescalate.yml:2]: wrong description

**Review by Arianna2028 — COMMENTED** [review]
This is pretty awesome. Some smaller things on the CircleCI stuff. The scripts seem reasonable but I'm not very confident in my ability to review them 😅

File: .circleci/test-deploy.yml [hunk 4]
```diff
@@ -7,22 +7,29 @@ filters: &filters
...
-    steps:
-      - checkout
-      # Run your orb's commands to validate them.
-      - approval/greet
 workflows:
   test-deploy:
     jobs:
-      # Make sure to include "filters: *filters" in every test job you want to run as part of your deployment.
-      - command-tests:
+      - approval/request:
```
Comment (komalali) [inline .circleci/test-deploy.yml:13]: Have we considered calling these `sym/request` and `sym/deescalate`? IMO it's a little clearer than `approval/deescalate`? And with other orbs I've generally seen the pattern of like `slack/notify` or `terraform/apply`. 

If at some point we do end up with multiple templates we can always publish a major version change with updated names. Curious what folks think.

Comment (komalali) [reply .circleci/test-deploy.yml:13]: answered in slack but yes `sym/sym` or `symops/sym` is fine

**Pull request closed without merging**